hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/BlockCache.java

```diff
@@ -170,4 +172,8 @@ default boolean isMetaBlock(BlockType blockType) {
   default Optional<Map<String, Boolean>> getFullyCachedFiles() {
     return Optional.empty();
   }
+
+  default List<HStoreFile> getCachedButClosedFiles() {
```

Comment:
I like the idea of exposing this info. 

And let's use Optional to avoid returning null anti-pattern, just like the getFullyCachedFiles method above. 

Also the naming is not very intuitive. We could call it getFilesWithStaleBlocks and let's put an explanatory javadoc.

Additionally, BlockCache interface should defined the clean method as well. It's up for the BlockCache implementation to deal with the eviction logic.